pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -115,15 +113,16 @@
     // Path to ZNode containing TimeAverageBrokerData jsons for each broker.
     public static final String TIME_AVERAGE_BROKER_ZPATH = "/loadbalance/broker-time-average";
 
-    // ZooKeeper Cache of the currently available active brokers.
-    // availableActiveBrokers.get() will return a set of the broker names without an http prefix.
-    private ZooKeeperChildrenCache availableActiveBrokers;
-
     // Set of broker candidates to reuse so that object creation is avoided.
     private final Set<String> brokerCandidateCache;
 
-    // ZooKeeper cache of the local broker data, stored in LoadManager.LOADBALANCE_BROKER_ROOT.
-    private ZooKeeperDataCache<LocalBrokerData> brokerDataCache;
+    // Cache of the local broker data, stored in LoadManager.LOADBALANCE_BROKER_ROOT.
+    private LockManager<LocalBrokerData> brokersData;
```

Comment:
Can we use MetadataCache instead LockManager? `LocalBrokerData` is a metadata stored in localzk right now and it doesn't need to acquire lock.

Reply:
It is a lock though, in the sense that we're acquiring an ephemeral z-node with the broker name. If there's already a z-node there we need to handle the scenario and that's what the ResourceLock does